awscli/customizations/s3/fileinfo.py

```diff
@@ -301,7 +301,7 @@ def delete(self):
         if (self.src_type == 's3'):
             bucket, key = find_bucket_key(self.src)
             params = {'Bucket': bucket, 'Key': key}
-            self.source_client.delete_object(**params)
+            self.client.delete_object(**params)
```

Comment:
Ahh now I remember why it is not simple as this. So this logic was put in place for the bucket to bucket ``mv`` command. When we move an object from one bucket to another, we first copy the object and then delete the other object. We must send the copy request to the destination region but the delete must happen from the source. If my assumption is correct, your PR will fail if you do a cross-region sigv4 enabled mv command. I think the change to get this fixed needs to happen in the [DeleteSync](https://github.com/aws/aws-cli/blob/develop/awscli/customizations/s3/syncstrategy/delete.py) strategy. Might be a little tricky though because a ``FileStat`` object does not know its client. I would also recommend looking in adding it in the ``FileInfoBuilder`` object as that is where the client gets injected.